src/gui/preferences.c

```diff
@@ -564,157 +582,163 @@ static void cairo_destroy_from_pixbuf(guchar *pixels, gpointer data)
-  dt_draw_cairo_to_gdk_pixbuf(data, DT_PIXEL_APPLY_DPI(ICON_SIZE), DT_PIXEL_APPLY_DPI(ICON_SIZE));
-  GdkPixbuf *lock_pixbuf = gdk_pixbuf_new_from_data(data, GDK_COLORSPACE_RGB, TRUE, 8,
-                                                    DT_PIXEL_APPLY_DPI(ICON_SIZE), DT_PIXEL_APPLY_DPI(ICON_SIZE),
-                                                    cairo_image_surface_get_stride(lock_cst),
-                                                    cairo_destroy_from_pixbuf, lock_cr);
-
-  // check mark
-  cairo_surface_t *check_cst = cairo_image_surface_create(CAIRO_FORMAT_ARGB32, DT_PIXEL_APPLY_DPI(ICON_SIZE),
-                                                          DT_PIXEL_APPLY_DPI(ICON_SIZE));
-  cairo_t *check_cr = cairo_create(check_cst);
-  cairo_set_source_rgb(check_cr, 0.7, 0.7, 0.7);
-  dtgtk_cairo_paint_check_mark(check_cr, 0, 0, DT_PIXEL_APPLY_DPI(ICON_SIZE), DT_PIXEL_APPLY_DPI(ICON_SIZE), 0, NULL);
-  cairo_surface_flush(check_cst);
-  data = cairo_image_surface_get_data(check_cst);
-  dt_draw_cairo_to_gdk_pixbuf(data, DT_PIXEL_APPLY_DPI(ICON_SIZE), DT_PIXEL_APPLY_DPI(ICON_SIZE));
-  GdkPixbuf *check_pixbuf = gdk_pixbuf_new_from_data(data, GDK_COLORSPACE_RGB, TRUE, 8,
-                                                     DT_PIXEL_APPLY_DPI(ICON_SIZE), DT_PIXEL_APPLY_DPI(ICON_SIZE),
-                                                     cairo_image_surface_get_stride(check_cst),
-                                                     cairo_destroy_from_pixbuf, check_cr);
-  // clang-format off
-  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db),
-                              "SELECT rowid, name, operation, autoapply, model, maker, lens, iso_min, "
-                              "iso_max, exposure_min, exposure_max, aperture_min, aperture_max, "
-                              "focal_length_min, focal_length_max, writeprotect FROM data.presets ORDER BY "
-                              "operation, name",
-                              -1, &stmt, NULL);
-  // clang-format on
-  while(sqlite3_step(stmt) == SQLITE_ROW)
+  const gint rowid = sqlite3_column_int(stmt, 0);
+  const gchar *name = (gchar *)sqlite3_column_text(stmt, 1);
+  const gchar *operation = (gchar *)sqlite3_column_text(stmt, 2);
+  const gboolean autoapply = (sqlite3_column_int(stmt, 3) == 0 ? FALSE : TRUE);
+  const gchar *model = (gchar *)sqlite3_column_text(stmt, 4);
+  const gchar *maker = (gchar *)sqlite3_column_text(stmt, 5);
+  const gchar *lens = (gchar *)sqlite3_column_text(stmt, 6);
+  const float iso_min = sqlite3_column_double(stmt, 7);
+  const float iso_max = sqlite3_column_double(stmt, 8);
+  const float exposure_min = sqlite3_column_double(stmt, 9);
+  const float exposure_max = sqlite3_column_double(stmt, 10);
+  const float aperture_min = sqlite3_column_double(stmt, 11);
+  const float aperture_max = sqlite3_column_double(stmt, 12);
+  const int focal_length_min = sqlite3_column_double(stmt, 13);
```

Comment:
Schema to be checked here. You get a double from the db and it is stored into an int. (I understand the code was this way before, but as we are at it better fixing if needed).